hack/generate-jobs.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generate job configuration files."""
+
+import argparse
+import configparser
+import filecmp
+import glob
+import os
+import shutil
+import sys
+import tempfile
+import typing
+
+import jinja2
+
+
+def generate(patterns: typing.List[str], only_verify: bool, overwrite: bool) -> str:
+    """
+    Generate job configuration files.
+    Return empty string if successful, error message otherwise.
+    """
+    for pattern in patterns:
+        paths = glob.glob(pattern)
+        if not paths:
+            return f"No files found for pattern {pattern}"
+        for fname in paths:
+            err = generate_one(fname, only_verify, overwrite)
+            if err:
+                return err
+    return ""
+
+
+def generate_one(fname: str, only_verify: bool, overwrite: bool) -> str:
+    config = configparser.ConfigParser()
+    config.read(fname)
+
+    template_name = config.get("DEFAULT", "template")
+    template_path = os.path.join(os.path.dirname(fname), template_name)
+    name = template_path.split(".")[0]
+    with open(template_path, encoding="utf-8") as inp:
+        template = jinja2.Template(inp.read(), lstrip_blocks=True)
+        for kind in config.get("DEFAULT", "kinds").split(","):
+            tmp = tempfile.NamedTemporaryFile(
+                "w",
+                prefix=os.path.splitext(os.path.basename(__file__))[0],
+                delete=False,
+            )
+            with tmp:
+                header = (
+                    "# GENERATED FILE - DO NOT EDIT!\n#\n# "
+                    "Instead, modify template and run `make generate-jobs`.\n"
```

Comment:
Someone not familiar with this mechanism will have to ask "which template?"

Let's answer that right away:
```suggestion
                    "Instead, modify %s and run `make generate-jobs`.\n" % os.path.basename(template_path)
```

I think dropping the path is okay because generated file and template will typically (always?) be in the same directory.

Reply:
fixed, thanks for pointing out